adsft/tests/test_tasks.py

```diff
@@ -181,6 +185,27 @@ def test_task_identify_facilities(self):
                     tasks.task_identify_facilities(msg)
                     # use logging to check logic here when we switch to python3
 
+    def test_task_apply_nlp(self):
+
+        msg = {
+                'bibcode': 'fta',
+                'file_format': 'pdf',
+                'meta_path': u'{}/ft/a/meta.json'.format(self.app.conf['FULLTEXT_EXTRACT_PATH']),
+                'acknowledgements': 'We thank the Alma team.',
+                'fulltext': 'Introduction\nTHIS IS AN INTERESTING TITLE\n'
+                }
+
+        with patch('adsft.writer.write_file', return_value=None) as task_write_text:
+            with patch('adsft.checker.load_meta_file', return_value=None) as load_meta:
+                with patch('adsft.reader.read_content', return_value=msg) as read_content:
+                    tasks.task_apply_nlp_technqiues(read_content())
+                    self.assertTrue(load_meta.called)
+                    self.assertTrue(read_content.called)
+                    self.assertTrue(task_write_text.called)
+                    actual = task_write_text.call_args[0]
+                    self.assertEqual(os.path.dirname(msg['meta_path'])+'/nlp.json', actual[0])
```

Comment:
The problem here is that I'm mocking spaCy for the facilities NER, so I can't mock one model and test the results of another. I can instead add a separate file for testing spacy's english model.

Reply:
After our meeting yesterday, I implemented your recommendation to check if the directory exists before calling spacy.load(). I realized I had tried this before, but ran into the same problem where 'en_core_web_sm' will be installed in different locations. This time I was able to resolve this using pkgutils. The unit tests will not pass now unless 'en_core_web_en' is downloaded because the model is used as a callable, so that's one problem if someone is developing locally and trying to run unit tests. In light of this, I think it would be a good idea to remove the installation of the model in eb-deploy using `python -m spacy download en_core_web_sm` and move that to the requirements.txt file in this PR by adding `https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.2.0/en_core_web_sm-2.2.0.tar.gz`. I found that this is also recommended in spaCy's documentation [here](https://spacy.io/usage/models#download-pip).